installer/internal/algo/common_ubuntu.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package algo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+)
+
+const (
+	// ImgpkgVersion defines the imgpkg version that will be installed on host if imgpkg is not already installed
+	ImgpkgVersion = "v0.36.4"
+)
+
+// BaseUbuntuInstaller provides common functionality for Ubuntu installers
+type BaseUbuntuInstaller struct {
+	install   string
+	uninstall string
+}
+
+// Install will return k8s install script
+func (s *BaseUbuntuInstaller) Install() string {
+	return s.install
+}
+
+// Uninstall will return k8s uninstall script
+func (s *BaseUbuntuInstaller) Uninstall() string {
+	return s.uninstall
+}
+
+// NewBaseUbuntuInstaller creates a new base Ubuntu installer
+func NewBaseUbuntuInstaller(ctx context.Context, arch, bundleAddrs string, containerdConfig string) (*BaseUbuntuInstaller, error) {
+	data := map[string]string{
+		"BundleAddrs":        bundleAddrs,
+		"Arch":               arch,
+		"ImgpkgVersion":      ImgpkgVersion,
+		"BundleDownloadPath": "{{.BundleDownloadPath}}",
+		"ContainerdConfig":   containerdConfig,
+	}
+
+	parseFn := func(script string) (string, error) {
+		parser, err := template.New("parser").Parse(script)
+		if err != nil {
+			return "", fmt.Errorf("unable to parse install script")
+		}
+		var tpl bytes.Buffer
+		if err = parser.Execute(&tpl, data); err != nil {
+			return "", fmt.Errorf("unable to apply install parsed template to the data object")
+		}
+		return tpl.String(), nil
+	}
+
+	install, err := parseFn(commonUbuntuInstallTemplate)
+	if err != nil {
+		return nil, err
+	}
+	uninstall, err := parseFn(commonUbuntuUninstallTemplate)
+	if err != nil {
+		return nil, err
+	}
+	return &BaseUbuntuInstaller{
+		install:   install,
+		uninstall: uninstall,
+	}, nil
+}
+
+// Common installation template for Ubuntu-based systems
+var commonUbuntuInstallTemplate = `
```

Comment:
Move it out into a separate template file.

Reply:
Sure...Doing that in my next commit